src/libs/auth.ts

```diff
@@ -232,7 +359,7 @@ export const processUser = (user, isSignInEvent) => {
     const userId = profile?.sub;
 
     // The following few lines of code are to handle sign-in failures due to privacy tools.
-    if (isSignInEvent === true && state.isSignedIn === false && isSignedIn === false) {
+    if (isSignInEvent === true && state.isSignedIn === false && !isSignedIn) {
```

Comment:
Nit: No need to change this. `isSignedIn` is a boolean, so `!isSignedIn` and `isSignedIn === false` are equivalent.

The mixture of checking exact values for `isSignInEvent` and `state.isSignedIn` vs "falsiness" for `isSignedIn` suggests that `isSignedIn` isn't a boolean.

Reply:
I would say that this is implying that `isSignedIn` is a boolean where as the others are not. Since the others are getting passed in through `AuthStoreSetter.js` I cannot guarantee that they are booleans, but I can guarantee that `isSignedIn` is a boolean. Ultimately, I think either is fine, but I did this because isSignedIn is a boolean so that exact value checking is not necessary. 

I will make a change to add explicit typing here. Let me know what you think.

Reply:
Terra UI has _a lot_ of code that uses `!value` to coerce a value to a boolean.

Admittedly, this is very much a nitpick, but it's the mix of patterns that's getting me. Using exact comparison (`value === true`, `value === false`) for some parts of the condition and `!value` for the last part leads me to believe there's some importance to why that last condition is different. It sounds like there's not though. So I'd suggest making it an exact comparison as well so it blends in with the others and doesn't distract the reader.